src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs

```diff
@@ -602,8 +646,17 @@ public virtual string Href([NotNull] string contentPath)
 
             BeginContext(attributeValue.Value.Position, sourceLength, isLiteral: attributeValue.Literal);
 
-            // The extra branching here is to ensure that we call the Write*To(string) overload where
-            // possible.
+            WriteUnprefixedAttributeValueTo(writer, attributeValue);
+
+            EndContext();
+        }
+
+        private void WriteUnprefixedAttributeValueTo(TextWriter writer, AttributeValue attributeValue)
```

Comment:
Eh, like the different method approach better than the unneeded parameter/two if statements to go along with it.